src/pip/_internal/utils/misc.py

```diff
@@ -710,6 +710,8 @@ def call_subprocess(
 
     logger.debug("Running command %s", command_desc)
     env = os.environ.copy()
+    if os.name == 'nt':
```

Comment:
I would lean towards doing this for all platforms (i.e. unconditionally) if it actually causes subprocess to break. Also, there is already code a few lines down to do this via the `unset_environ` variable. So you could just make a copy of the `unset_environ` list towards the beginning of this function and add the empty string to it if it's not already present.

Reply:
Hi @cjerdonek does `test_utils_subprocess.py` look like the right place to add this ?

Reply:
Sorry for the delayed response here: yes.

Reply:
Cheers, I've had a lot more unit test experience in the meantime, will try and sort this in the next few days.

Reply:
OK, finally got some time to give this a go - there were some subtleties getting this working.

Reply:
Monkeypatching the setenviron didn't seem to work, so I settled for passing the environment var to extra_environ, along with another sentinel value (to verify vars are getting set at all).